core/server/namespace.go

```diff
@@ -37,3 +38,26 @@ func (as *coreServer) GetFluxNamespace(ctx context.Context, msg *pb.GetFluxNames
 
 	return &pb.GetFluxNamespaceResponse{Name: nsList.Items[0].Name}, nil
 }
+
+func (as *coreServer) ListNamespaces(ctx context.Context, msg *pb.ListNamespacesRequest) (*pb.ListNamespacesResponse, error) {
+	k8s, err := as.k8s.Client(ctx)
+	if err != nil {
+		return nil, doClientError(err)
```

Comment:
Small tip, `doClientError` 

```go
return status.Errorf(codes.Internal, "unable to make k8s rest client: %s", err.Error())
```
From https://pkg.go.dev/fmt
>If the format (which is implicitly %v for Println etc.) is valid for a string (%s %q %v %x %X), the following two rules apply:
>
>4. If an operand implements the error interface, the Error method will be invoked to convert the object to a string, which will then be formatted as required by the verb (if any).

Reply:
I have no idea what to do with this comment. I had no intention to change what we are using, if it's not a refactor PR. I like not to combine random things in a single PR if they are not strictly related, that makes the process to revert changes much easier.

As a totally personal opinion, even if I wrote that part, I would still use the `Error()` function on it, because we don't know how long this code will live there, it will work no matter what happens and explicitly called a function that returns with a string. In that case, we have a nice and unified way to represent errors. No one can guarantee the the order of the checks will not change, or no one will implement (don't ask why, ppl can be weird) Formatter on an error type, in which case fmt will use that instead of the Error function.